### PR TITLE
chore: docker-compose templatization

### LINKED ISSUE #531: Templatization of our docker-compose files
Today we maintain 4 different docker-compose files:

- docker-compose.yml
- docker-compose.dev.yml
- docker-compose-with-otel.yaml
- docker-compose-with-auth.yml

The main difference between them is either extra services or different environment variables.
We need to maintain 1 "baseline" docker-compose file that other docker-compsoe files inherit and extend.

### PR DESCRIPTION
**Pull request by mohanish2504**
/close #531

- Templatization docker compose file
- Fixes Dockerfile.dev.api entry point
so that also calls config.py at the start of gunicorn server.
- Updates volume binding for backend in otel compose file

### CODE REVIEW

File: docker/Dockerfile.dev.api [hunk 1]
```diff
@@ -25,4 +25,4 @@ RUN echo "test"
 ENV PATH="/venv/bin:${PATH}"
 ENV VIRTUAL_ENV="/venv"
 
-CMD ["uvicorn", "keep.api.api:get_app", "--host", "0.0.0.0", "--port", "8080", "--reload"]
```
Comment (talboren) [inline docker/Dockerfile.dev.api:28]: `--reload` is required

**Comment by talboren** [comment]
@mohanish2504 this looks awesome. have you tested different docker-compose files and made sure that it works properly? (e.g. with-auth vs. no-auth)

**Comment by mohanish2504** [comment]
@talboren all the 4 files are tested.

**Comment by mohanish2504** [comment]
@talboren added reload options.
Ran a test on the dev & otel docker-compose file, and it is good to go.

**Pull request merged**